### DISCUSSION TITLE
Recommend native overlayfs instead of fuse-overlayfs?

### DISCUSSION THREAD

**Discussion by StarpTech** [containers/podman#15465] [General]
Hi, according to https://www.redhat.com/sysadmin/podman-rootless-overlay is it still recommended to use `mount_program = "/usr/bin/fuse-overlayfs"` when Kernel 5.13 has shipped native support?
Anything else to consider except clearing the container storage and comment `# mount_program = "/usr/bin/fuse-overlayfs"`?

https://github.com/containers/podman/blob/34d516840df73787d2359037e4a2ab51849e59eb/contrib/podmanimage/stable/Containerfile#L38-L43

**Comment by mheon** [comment]
I don't think that blog says we still recommend `fuse-overlayfs` when native Overlay is available; we prefer native overlay unless you have a good reason to continue to use fuse-overlay (it does some things we don't have kernel support for, at least last I checked; UID mapping would be the most notable).

Commenting the `mount_program` line will indeed be sufficient to swap the driver.

**Reply by rhatdan** [reply]
The problem we have is changing this now, means that older versions of Podman will break.

At 5.0 we should probably make the change to drop fuse-overlayfs from inside of the container.  Or can we do this now @giuseppe

**Reply by vrothberg** [reply]
Isn't c/storage clever enough to figure out using native overlayfs on its own?

**Reply by StarpTech** [reply]
Yes, it should if we comment out the driver overwrite. My question is more from a consumer perspective. I'm not familiar with all the internals. I just want to set it up in the best way :)

**Reply by rhatdan** [reply]
Open a PR to make the change then we can see if anyone on older OS compain.

**Comment by StarpTech** [comment]
>  (it does some things we don't have kernel support for, at least last I checked; UID mapping would be the most notable).

This is why asked. I wanted to verify if the native overlay is a drop-in replacement. I couldn't find anything about this in the repository. All examples use the fuse overlay. Thanks!
